messenger_test.go

```diff
@@ -150,6 +130,250 @@ func (s *MessengerSuite) TestRetrievePrivate() {
 	s.Equal(&s.privateKey.PublicKey, message.SigPubKey) // this is OUR message
 }
 
+func (s *MessengerSuite) TestChatPersistencePublic() {
+	chat := Chat{
+		ID:                     "chat-name",
+		Name:                   "chat-name",
+		Color:                  "#fffff",
+		Active:                 true,
+		ChatType:               ChatTypePublic,
+		Timestamp:              10,
+		LastClockValue:         20,
+		DeletedAtClockValue:    30,
+		UnviewedMessagesCount:  40,
+		LastMessageContentType: "something",
+		LastMessageContent:     "something-else",
+	}
+
+	s.Require().NoError(s.m.SaveChat(chat))
+	savedChats, err := s.m.GetChats(0, 10)
+	s.Require().NoError(err)
+	s.Require().Equal(1, len(savedChats))
+
+	actualChat := savedChats[0]
+
+	s.Require().Equal(chat.ID, actualChat.ID)
```

Comment:
```go
s.Equal(chat, savedChats[0])
```

should have the same effect or this if we have some additional fields not present or having different expected values:
```go
expectedChat := chat
expectedChat.SomeField = SomeValue
s.Equal(expectedChat, savedChats[0])
```